src/home/rooms_list.rs

```diff
@@ -475,6 +482,19 @@ impl Widget for RoomsList {
                             error!("Error: couldn't find room {room_id} to update latest event");
                         }
                     }
+                    RoomsListUpdate::UpdateNumUnreadMessages { room_id, count } => {
+                        self.all_rooms
+                            .get_mut(&room_id)
+                            .map(|room| {
+                                room.num_unread_messages = match count {
+                                    UnreadMessageCount::Unknown => 0,
+                                    UnreadMessageCount::Known(count) => count,
+                                };
+                            })
+                            .unwrap_or_else(|| {
+                                error!("Error: couldn't find room {} to update unread messages count", room_id);
+                            })
```

Comment:
the `.map` combinator should only be used to modify a value before returning it, not to sneakily perform side effects like this. I know it's tempting to do this, but it's clearer and more canonical to use an `if ... else` block here, like so:
```suggestion
                        if let Some(room) = self.all_rooms.get_mut(&room_id) {
                            room.num_unread_messages = match count {
                                UnreadMessageCount::Unknown => 0,
                                UnreadMessageCount::Known(count) => count,
                            };
                        } else {
                            error!("Error: couldn't find room {} to update unread messages count", room_id);
                        }
```